xarray/core/rolling.py

```diff
@@ -64,34 +86,38 @@ def __init__(self, obj, windows, min_periods=None, center=False):
         windows : mapping of hashable to int
             A mapping from the name of the dimension to create the rolling
             window along (e.g. `time`) to the size of the moving window.
-        min_periods : int, default: None
+        min_periods : int or None, default: None
             Minimum number of observations in window required to have a value
             (otherwise result is NA). The default, None, is equivalent to
             setting min_periods equal to the size of the window.
-        center : bool, default: False
-            Set the labels at the center of the window.
+        center : bool or dict-like Hashable to bool, default: False
+            Set the labels at the center of the window. If dict-like, set this
+            property per rolling dimension.
 
         Returns
         -------
         rolling : type of input argument
         """
-        self.dim, self.window = [], []
+        self.dim: list[Hashable] = []
+        self.window: list[int] = []
         for d, w in windows.items():
             self.dim.append(d)
             if w <= 0:
                 raise ValueError("window must be > 0")
             self.window.append(w)
 
         self.center = self._mapping_to_list(center, default=False)
-        self.obj = obj
+        self.obj: T_Xarray = obj
 
         # attributes
         if min_periods is not None and min_periods <= 0:
             raise ValueError("min_periods must be greater than zero or None")
 
-        self.min_periods = np.prod(self.window) if min_periods is None else min_periods
+        self.min_periods = (
+            int(np.prod(self.window)) if min_periods is None else min_periods
```

Comment:
`math.prod` is a clean alternative here:
```suggestion
            math.prod(self.window) if min_periods is None else min_periods
```